projects/ion/src/lib/step/step.component.ts

```diff
@@ -0,0 +1,131 @@
+import {
+  Component,
+  EventEmitter,
+  Input,
+  OnChanges,
+  OnInit,
+  Output,
+  SimpleChanges,
+} from '@angular/core';
+
+export type StatusType = 'default' | 'selected' | 'checked' | 'error';
+
+export interface StepType {
+  label: string;
+  description?: string;
+  index?: number;
+  status?: StatusType;
+}
+
+export type StepTheme = 'default' | 'pastel-tones' | 'dark-tones';
+
+export type CrossType = 'initial' | 'final';
+
+export type StepConfig = {
+  current: number;
+  disabled?: boolean;
+  theme?: StepTheme;
+  steps: StepType[];
+  clickable?: boolean;
+};
+
+enum Status {
+  default = 'default',
+  selected = 'selected',
+  checked = 'checked',
+  error = 'error',
+}
+
+@Component({
+  selector: 'ion-step',
+  templateUrl: './step.component.html',
+  styleUrls: ['./step.component.scss'],
+})
+export class StepComponent implements OnInit, OnChanges {
+  @Input() current = 1;
+  @Input() disabled = false;
+  @Input() theme: StepTheme = 'default';
+  @Input() steps: StepType[];
+  @Input() clickable: boolean;
+  @Output() indexChange = new EventEmitter<number>();
+
+  public firstCatchStatus = true;
+
+  public crossVisibility(crossType: CrossType, step: StepType): boolean {
+    const condition =
+      crossType === 'initial'
+        ? step.index === 1
+        : step.index === this.steps.length;
+
+    return !condition;
+  }
+
+  public beforeStepIsChecked(index: number): boolean {
+    return this.steps[index - 1] && this.steps[index - 1].status === 'checked';
+  }
+
+  public afterStepIsChecked(index: number): boolean {
+    return (
+      this.steps[index + 1] &&
+      (this.steps[index + 1].status === Status.checked ||
+        this.steps[index + 1].status === Status.selected ||
+        this.steps[index + 1].status === Status.error)
+    );
+  }
```

Comment:
```suggestion
  public afterStepIsDefault(index: number): boolean {
    return (
      this.steps[index + 1] && this.steps[index + 1].status === Status.default
    );
  }
```
I can be wrong, but I understand this function as checking if status is default. You can invert this logic to check if status is default and changing your HTML to behave as you want.

Reply:
Actually, it's checking if next step is everything, unless default. So u got me an idea, and I'll check for it like this: this.steps[index + 1] && this.steps[index + 1].status !== Status.default